src/mongocxx/options/encrypt.cpp

```diff
@@ -79,33 +81,83 @@ const stdx::optional<options::range>& encrypt::range_opts() const {
     return _range_opts;
 }
 
+namespace {
+
+struct scoped_bson_value {
+    bson_value_t value = {};
+
+    // Allow obtaining a pointer to this->value even in rvalue expressions.
+    bson_value_t* get() noexcept {
+        return &value;
+    }
+
+    // Ensure this->value can be initialized/overwritten properly.
+    bson_value_t* value_for_init() noexcept {
+        bson_value_destroy(&value);
```

Comment:
As implemented, `bson_value_destroy` is not safe to call repeatedly without resetting the `value_type`. Calling repeatedly could result in a double free.

If `value_for_init` is intended to be safe to call repeatedly, suggest zeroing `value` with `memset(&value, 0, sizeof(bson_value_t));` or equivalent.

Reply:
It is probably excessive accomodation to support reinitialization of `scoped_bson_value`. I will opt to remove code that implies reinitialization is supported.